notebooks/config.ipynb

```diff
@@ -964,12 +971,831 @@
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 32,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.compare(\n",
+    "    config_1_path='fixtures/config/base-config.nml',\n",
+    "    config_2_path='tmp/config-copy.nml',\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e14c0158-baa4-45d0-a86c-d9a245021229",
+   "metadata": {},
+   "source": [
+    "## Validating Configs\n",
+    "\n",
+    "The `config.validate()` function checks if a given config conforms to a specified JSON schema.\n",
+    "<!--cell 64-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "id": "2eb5b2ce-bebd-449d-90a5-764484aa03aa",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Help on function validate in module uwtools.api.config:\n",
+      "\n",
+      "validate(schema_file: Union[pathlib.Path, str], config: Union[dict, str, uwtools.config.formats.yaml.YAMLConfig, pathlib.Path, NoneType] = None, stdin_ok: bool = False) -> bool\n",
+      "    Check whether the specified config conforms to the specified JSON Schema spec.\n",
+      "\n",
+      "    If no config is specified, ``stdin`` is read and will be parsed as YAML and then validated. A\n",
+      "    ``dict`` or a YAMLConfig instance may also be provided for validation.\n",
+      "\n",
+      "    :param schema_file: The JSON Schema file to use for validation.\n",
+      "    :param config: The config to validate.\n",
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: ``True`` if the YAML file conforms to the schema, ``False`` otherwise.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.validate)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1eeca5ad-e969-4670-bddd-8e20b38f3d6c",
+   "metadata": {},
+   "source": [
+    "Below is an example of a schema used to validate a config. It ensures that the required keys are present and the value types match expectations.\n",
+    "<!--cell 66-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "id": "f9c66a24-2a7e-43be-839f-3ad5b136b646",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"additionalProperties\": false,\n",
+      "  \"properties\": {\n",
+      "    \"greeting\": {\n",
+      "      \"type\": \"string\"\n",
+      "    },\n",
+      "    \"recipient\": {\n",
+      "      \"type\": \"string\"\n",
+      "    }\n",
+      "  },\n",
+      "  \"required\": [\n",
+      "    \"greeting\", \"recipient\"\n",
+      "  ],\n",
+      "  \"type\": \"object\"\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/config/validate.jsonschema"
```

Comment:
Does it make sense to add a reminder cell showing what's in the `get-config.yaml` that this schema is going to be validating since we've more recently seen a couple of other YAML contents?

Reply:
Yeah I think it's worthwhile to show this file here. I added a code block before this one to do so.